### PR TITLE
Improve logs

### PR DESCRIPTION
**Pull request by anxolin**
It takes care of a few issues:

- Adds more context to the log messages, so we know which component is writing
- Some `catch` exception logics were using the normal log function. Now it's using `console.error`
- `log` function was renamed to `logInfo`, it has the same logic
- `logDebug` is a new function that is hidden if we are not in develop (by default)
- We can enable debug by executing the global function `window.toggleDebug()`

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr576--dexreact.review.gnosisdev.com

File: src/components/DepositWidget/useRowActions.tsx [hunk 1]
```diff
@@ -140,11 +140,11 @@ export const useRowActions = (params: Params): Result => {
...
       const receipt = await depositApi.withdraw({ userAddress, tokenAddress, networkId, txOptionalParams })
 
-      log(`The transaction has been mined: ${receipt.transactionHash}`)
+      logDebug(`[DepositWidget:useRowActions] The transaction has been mined: ${receipt.transactionHash}`)
       toast.success(`Withdraw of ${formatAmount(pendingWithdraw.amount, decimals)} ${symbol} completed`)
     } catch (error) {
-      console.error('Error executing the withdraw request', error)
-      toast.error(`Error executing the withdraw request: ${error.message}`)
+      console.error('[DepositWidget:useRowActions] Error executing the withdraw request', error)
+      toast.error(`[DepositWidget:useRowActions] Error executing the withdraw request: ${error.message}`)
```
Comment (alfetopito) [inline src/components/DepositWidget/useRowActions.tsx]: Should we show that internal info in the toast notification?

File: src/hooks/useTokenBalances.ts [hunk 2]
```diff
@@ -75,8 +75,8 @@ async function _getBalances(walletInfo: WalletInfo): Promise<TokenBalanceDetails
 
   const balancePromises: Promise<TokenBalanceDetails | null>[] = tokens.map(token =>
     fetchBalancesForToken(token, userAddress, contractAddress, networkId).catch(e => {
-      log('Error for', token, userAddress, contractAddress)
-      log(e)
+      console.error('[useTokenBalances] Error for', token, userAddress, contractAddress)
+      console.error(e)
```
Comment (alfetopito) [inline src/hooks/useTokenBalances.ts]: What about logging both into a single log entry?

File: src/services/factories/addTokenToExchange.ts [hunk 3]
```diff
@@ -55,9 +55,9 @@ export function addTokenToExchangeFactory(
       }
 
       // TODO: cache new token
-      log('new token: ', token)
+      logDebug('[services:factories:addTokenToExchange] New token: ', token)
     } catch (e) {
-      log('Failed to add new token (%s) to local list of tokens', tokenAddress, e)
+      console.error('Failed to add new token (%s) to local list of tokens', tokenAddress, e)
```
Comment (alfetopito) [inline src/services/factories/addTokenToExchange.ts]: Add the `[whatever]` prefix to this message

File: src/components/DepositWidget/useRowActions.tsx (hunk 1 above)
Comment (anxolin) [reply src/components/DepositWidget/useRowActions.tsx]: Ups! good thing you saw that :)

File: src/hooks/useTokenBalances.ts (hunk 2 above)
Comment (anxolin) [reply src/hooks/useTokenBalances.ts]: sure

File: src/utils/miscellaneous.ts [hunk 4]
```diff
@@ -13,7 +13,28 @@ export function assertNonNull<T>(val: T, message: string): asserts val is NonNul
...
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+let debugEnabled = process.env.NODE_ENV === 'development'
+
+declare global {
+  interface Window {
+    toggleDebug: () => void
+  }
+}
+
+window.toggleDebug = (): boolean => {
```
Comment (Velenir) [inline src/utils/miscellaneous.ts]: Do we wan't this available in production too?

Comment (anxolin) [reply src/utils/miscellaneous.ts]: Yes, that's the main reason I added that. It allows us to see the message we see today in a minimized version, only difference, logs are disabled by default

Comment (Velenir) [reply src/utils/miscellaneous.ts]: Cool then

File: src/utils/miscellaneous.ts [hunk 5]
```diff
@@ -13,7 +13,28 @@ export function assertNonNull<T>(val: T, message: string): asserts val is NonNul
...
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export const log = process.env.NODE_ENV === 'test' ? noop : (...args: any[]): void => console.log(...args)
+export const logInfo = process.env.NODE_ENV === 'test' ? noop : (...args: any[]): void => console.log(...args)
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+let debugEnabled = process.env.NODE_ENV === 'development'
+
+declare global {
+  interface Window {
+    toggleDebug: () => void
```
Comment (Velenir) [inline src/utils/miscellaneous.ts]: this should be
```suggestion
    toggleDebug: () => boolean
```
no?

Comment (anxolin) [reply src/utils/miscellaneous.ts]: I had made that change in a different PR. So it's now in this one (I've merged develop here)

**Pull request merged**